backend/sites/src/app/services/site/site.service.ts

```diff
@@ -1136,6 +1142,57 @@ export class SiteService {
     }
   }
 
+  /**
+   * Processes and saves site disclosure based on the provided actions.
+   * @param siteDisclosure - Site disclosure data including actions to be performed.
+   * @param userInfo - Information about the user performing the actions.
+   * @param transactionalEntityManager - Entity manager for handling transactions.
+   */
+  async processSiteDisclosure(
+    siteDisclosure: any[],
+    userInfo: any,
+    transactionalEntityManager: EntityManager,
+  ) {
+    if (siteDisclosure && siteDisclosure.length > 0) {
+      const { apiAction, id, ...disclosureData } = siteDisclosure[0];
+      let profile = {
+        ...new SiteProfiles(),
```

Comment:
I would recommend to remove the logic where only accessing specific index. So either process the loop or change it to accept only single object

Reply:
done